uniffi_meta/src/lib.rs

```diff
@@ -174,6 +174,27 @@ impl MethodMetadata {
     }
 }
 
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Deserialize, Serialize, Checksum)]
+pub enum UniffiTrait {
+    Debug,
+    Display,
+    PartialEq,
+    Hash,
+}
+
+impl TryFrom<&str> for UniffiTrait {
+    type Error = anyhow::Error;
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        match value {
+            "Debug" => Ok(UniffiTrait::Debug),
+            "Display" => Ok(UniffiTrait::Display),
+            "PartialEq" => Ok(UniffiTrait::PartialEq),
```

Comment:
Should this be `Eq`?  It seems weird to define the `==` operator, but not have it transitive/reflexive/symmetric.  I'm not really sure about this one though.

Reply:
Me either - PartialEq is what actually defines the methods we use though, but I guess `Eq` is probably closer to what other languages actually need here (at least, is a ore conservative choice and likely to support all languages). So yeah, I guess `Eq` is better

Reply:
To put it another way: if a struct defines `PartialEq` but not `Eq`, and the definition isn't transitive, IIUC that would meet the requirement for rust, but unlikely to meet the requirement for foreign languages. Rust implementing `Eq` would.